src/bentoml/_internal/runner/runner_handle/local.py

```diff
@@ -61,3 +63,19 @@ async def async_run_method(
             *args,
             limiter=self._limiter,
         )
+
+    async def async_stream_method(
+        self,
+        __bentoml_method: RunnerMethod[t.Any, P, R],
+        *args: P.args,
+        **kwargs: P.kwargs,
+    ) -> AsyncGenerator[R, None]:
+        generator = getattr(self._runnable, __bentoml_method.name)(*args, **kwargs)
+
+        # This allows user to define both sync and async generator
+        if inspect.isasyncgen(generator):
+            async for chunk in generator:
+                yield chunk
+        else:
+            for chunk in generator:
+                yield chunk
```

Comment:
This concerns me a bit, doesn't this mean that we're blocking the async thread?

Reply:
yeah, it's not really converting sync -> async for generators as we were doing for the `async_run` and `run`

Reply:
Let's do that?

Reply:
[PEP 492 has an example](https://peps.python.org/pep-0492/#example-2) for getting async generator from a generator. But it looks the same as this approach?

Reply:
I suspect this problem is why the example says "While this is not a very useful thing to do..."

It may be that we want to follow django's example and simply disallow async generators; we may want to measure perf overhead here.

Reply:
if that is the case, then we should just disallow sync generators no? django

I'm thinking to maybe further improve this in another PR as there's additional intricacies to cast sync generators to a async via a threadpool, and like you said, we may want to do some benchmarking.

Reply:
Let's disallow this for now then.

Reply:
@aarnphm ^